Sources/SwiftKafka/KafkaProducer.swift

```diff
@@ -13,22 +13,183 @@
 //===----------------------------------------------------------------------===//
 
 import Crdkafka
+import struct Foundation.UUID
 import Logging
 
 /// Send messages to the Kafka cluster.
 /// - Note: When messages get published to a non-existent topic, a new topic is created using the ``KafkaTopicConfig``
 /// configuration object (only works if server has `auto.create.topics.enable` property set).
-public struct KafkaProducer {
+public actor KafkaProducer {
+    /// Class that lets us wrap both the `messageID` and the `producer` into a single pointer
+    private final class CallbackOpaquePointer {
+        let messageID: UUID // TODO: to avoid Foundation, have a counter in KafkaProducer that serves as ID
```

Comment:
I went with `UUID` here, but if we want to stay away from `Foundation` I could imagine having an isolated `counter` variable that can get passed as a message-id.